src/com/google/enterprise/gsafeed/Acl.java

```diff
@@ -70,24 +106,45 @@ public Acl setUrl(String value) {
   /**
    * Gets the value of the inheritanceType property.
    *
-   * @return possible object is {@link String}
+   * @return possible object is {@link InheritanceType}
    */
-  public String getInheritanceType() {
+  public InheritanceType getInheritanceType() {
     if (inheritanceType == null) {
-      return "leaf-node";
+      return null;
     } else {
-      return inheritanceType;
+      return InheritanceType.fromString(inheritanceType);
     }
   }
 
   /**
    * Sets the value of the inheritanceType property.
    *
-   * @param value allowed object is {@link String}
+   * @param value allowed object is {@link String} or null
    * @return this object
+   * @throws IllegalArgumentException if value isn't a valid
+   * inheritance-type attribute value
    */
   public Acl setInheritanceType(String value) {
-    this.inheritanceType = value;
+    if (Strings.isNullOrEmpty(value)) {
+      this.inheritanceType = null;
+    } else {
+      setInheritanceType(InheritanceType.fromString(value));
+    }
+    return this;
+  }
+
+  /**
+   * Sets the value of the inheritanceType property.
+   *
+   * @param value allowed object is {@link InheritanceType} or null
+   * @return this object
+   */
+  public Acl setInheritanceType(InheritanceType value) {
+    if (value == null) {
```

Comment:
If we keep this string representation, this code block is pretty repetitive. Maybe:

    this.inheritanceType = (value == null) ? null : value.toString();
or

    this.inheritanceType = toString(value, null);

where toString is an implementation of Objects.toString (which we can't use; it's Java 7).